core/commonTest/src/kotlinx/serialization/features/ThirdPartyGenericsTest.kt

```diff
@@ -0,0 +1,70 @@
+package kotlinx.serialization.features
+
+import kotlinx.serialization.KSerializer
+import kotlinx.serialization.Serializable
+import kotlinx.serialization.descriptors.SerialDescriptor
+import kotlinx.serialization.encoding.Decoder
+import kotlinx.serialization.encoding.Encoder
+import kotlinx.serialization.modules.SerializersModule
+import kotlinx.serialization.modules.contextual
+import kotlinx.serialization.serializer
+import kotlin.test.Test
+import kotlin.test.assertEquals
+
+open class ThirdPartyGenericsTest {
+    // This is a 3rd party class that we can't annotate as @Serializable
+    data class ThirdPartyBox<T>(val contents: T)
+
+    // This is the item that we put in the ThirdPartyBox, we control it, so can annotate it
+    @Serializable
+    data class Item(val name: String)
+
+    // The serializer for the ThirdPartyBox<T>
+    class BoxSerializer<T>(dataSerializer: KSerializer<T>) : KSerializer<ThirdPartyBox<T>> {
+        @Serializable
+        data class BoxSurrogate<T>(val contents: T)
+
+        private val strategy = BoxSurrogate.serializer(dataSerializer)
+        override val descriptor: SerialDescriptor = strategy.descriptor
+
+        override fun deserialize(decoder: Decoder): ThirdPartyBox<T> {
+            return ThirdPartyBox(decoder.decodeSerializableValue(strategy).contents)
+        }
+
+        override fun serialize(encoder: Encoder, value: ThirdPartyBox<T>) {
+            encoder.encodeSerializableValue(strategy, BoxSurrogate(value.contents))
+        }
+    }
+
+    // Register contextual serializer for ThirdPartyBox<Item>
+    protected val boxWithItemSerializer = BoxSerializer(Item.serializer())
+    protected val serializersModule = SerializersModule {
+        contextual(boxWithItemSerializer)
+    }
+
+    @Test
+    fun testSurrogateSerializerFoundForGenericWithKotlinType() {
+        /*
+         * This used to fail, because SerializersModule.serializerByKTypeImpl() never tried `getContextual()` after
```

Comment:
This comment is kinda excessive, because a) you explained bug in the commit message b) 'Fixed by changing `SerializersModule.serializerByKTypeImpl()` line 117' is literally the change in the commit. So I suggest to remove it.

Reply:
Yes, indeed. Left it in on purpose if any discussion would have followed. Will remove now.